// primeng/esm2022/config/themeprovider.mjs
import { DOCUMENT } from '@angular/common';
import { effect, inject, Injectable, signal, untracked } from '@angular/core';
import { Theme, ThemeService } from '@primeuix/styled';
import { BaseStyle } from 'primeng/base';
import * as i0 from "@angular/core";
export class ThemeProvider {
    // @todo define type for theme
    theme = signal(undefined);
    isThemeChanged = false;
    document = inject(DOCUMENT);
    baseStyle = inject(BaseStyle);
    constructor() {
        effect(() => {
            ThemeService.on('theme:change', (newTheme) => {
                untracked(() => {
                    this.isThemeChanged = true;
                    this.theme.set(newTheme);
                    // this.onThemeChange(this.theme());
                });
            });
        }, { allowSignalWrites: true });
        effect(() => {
            const themeValue = this.theme();
            if (this.document && themeValue) {
                if (!this.isThemeChanged) {
                    this.onThemeChange(themeValue);
                }
                this.isThemeChanged = false;
            }
        });
    }
    ngOnDestroy() {
        Theme.clearLoadedStyleNames();
        ThemeService.clear();
    }
    onThemeChange(value) {
        Theme.setTheme(value);
        if (this.document) {
            this.loadCommonTheme();
        }
    }
    loadCommonTheme() {
        if (this.theme() === 'none')
            return;
        // common
        if (!Theme.isStyleNameLoaded('common')) {
            const { primitive, semantic, global, style } = this.baseStyle.getCommonTheme?.() || {};
            const styleOptions = { nonce: undefined };
            this.baseStyle.load(primitive?.css, { name: 'primitive-variables', ...styleOptions });
            this.baseStyle.load(semantic?.css, { name: 'semantic-variables', ...styleOptions });
            this.baseStyle.load(global?.css, { name: 'global-variables', ...styleOptions });
            this.baseStyle.loadTheme({ name: 'global-style', ...styleOptions }, style);
            Theme.setLoadedStyleName('common');
        }
    }
    setThemeConfig(config) {
        const { theme } = config || {};
        if (theme)
            this.theme.set(theme);
    }
    static ɵfac = function ThemeProvider_Factory(__ngFactoryType__) { return new (__ngFactoryType__ || ThemeProvider)(); };
    static ɵprov = /*@__PURE__*/ i0.ɵɵdefineInjectable({ token: ThemeProvider, factory: ThemeProvider.ɵfac, providedIn: 'root' });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(ThemeProvider, [{
        type: Injectable,
        args: [{ providedIn: 'root' }]
    }], () => [], null); })();
