// primeng/esm2022/inputotp/style/inputotpstyle.mjs
import { Injectable } from '@angular/core';
import { BaseStyle } from 'primeng/base';
import * as i0 from "@angular/core";
const theme = ({ dt }) => `
.p-inputotp {
    display: flex;
    align-items: center;
    gap: 0.5rem;
}

.p-inputotp-input {
    text-align: center;
    width: 2.5rem;
}

.p-inputotp-input.p-inputtext-sm {
    text-align: center;
    width: ${dt('inputotp.input.sm.width')};
}

.p-inputotp-input.p-inputtext-lg {
    text-align: center;
    width: ${dt('inputotp.input.lg.width')};
}
`;
const classes = {
    root: 'p-inputotp p-component',
    pcInput: 'p-inputotp-input'
};
export class InputOtpStyle extends BaseStyle {
    name = 'inputotp';
    theme = theme;
    classes = classes;
    static ɵfac = /*@__PURE__*/ (() => { let ɵInputOtpStyle_BaseFactory; return function InputOtpStyle_Factory(__ngFactoryType__) { return (ɵInputOtpStyle_BaseFactory || (ɵInputOtpStyle_BaseFactory = i0.ɵɵgetInheritedFactory(InputOtpStyle)))(__ngFactoryType__ || InputOtpStyle); }; })();
    static ɵprov = /*@__PURE__*/ i0.ɵɵdefineInjectable({ token: InputOtpStyle, factory: InputOtpStyle.ɵfac });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(InputOtpStyle, [{
        type: Injectable
    }], null, null); })();
/**
 *
 * InputOtp is used to enter one time passwords.
 *
 * [Live Demo](https://www.primeng.org/inputotp/)
 *
 * @module inputotpstyle
 *
 */
export var InputOtpClasses;
(function (InputOtpClasses) {
    /**
     * Class name of the root element
     */
    InputOtpClasses["root"] = "p-inputotp";
    /**
     * Class name of the input element
     */
    InputOtpClasses["pcInput"] = "p-inputotp-input";
})(InputOtpClasses || (InputOtpClasses = {}));
