// primeng/esm2022/divider/divider.mjs
import { CommonModule } from '@angular/common';
import { ChangeDetectionStrategy, Component, HostBinding, inject, Input, NgModule, ViewEncapsulation } from '@angular/core';
import { SharedModule } from 'primeng/api';
import { BaseComponent } from 'primeng/basecomponent';
import { DividerStyle } from './style/dividerstyle';
import * as i0 from "@angular/core";
const _c0 = ["*"];
/**
 * Divider is used to separate contents.
 * @group Components
 */
export class Divider extends BaseComponent {
    /**
     * Inline style of the component.
     * @group Props
     */
    style;
    /**
     * Style class of the component.
     * @group Props
     */
    styleClass;
    /**
     * Specifies the orientation.
     * @group Props
     */
    layout = 'horizontal';
    /**
     * Border style type.
     * @group Props
     */
    type = 'solid';
    /**
     * Alignment of the content.
     * @group Props
     */
    align;
    _componentStyle = inject(DividerStyle);
    get hostClass() {
        return this.styleClass;
    }
    static ɵfac = /*@__PURE__*/ (() => { let ɵDivider_BaseFactory; return function Divider_Factory(__ngFactoryType__) { return (ɵDivider_BaseFactory || (ɵDivider_BaseFactory = i0.ɵɵgetInheritedFactory(Divider)))(__ngFactoryType__ || Divider); }; })();
    static ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: Divider, selectors: [["p-divider"]], hostVars: 35, hostBindings: function Divider_HostBindings(rf, ctx) { if (rf & 2) {
            i0.ɵɵattribute("aria-orientation", ctx.layout)("data-pc-name", "divider")("role", "separator");
            i0.ɵɵstyleMap(ctx.inlineStyles);
            i0.ɵɵclassMap(ctx.hostClass);
            i0.ɵɵstyleProp("justify-content", ctx.layout === "horizontal" ? ctx.align === "center" || ctx.align === undefined ? "center" : ctx.align === "left" ? "flex-start" : ctx.align === "right" ? "flex-end" : null : null)("align-items", ctx.layout === "vertical" ? ctx.align === "center" || ctx.align === undefined ? "center" : ctx.align === "top" ? "flex-start" : ctx.align === "bottom" ? "flex-end" : null : null);
            i0.ɵɵclassProp("p-divider", true)("p-component", true)("p-divider-horizontal", ctx.layout === "horizontal")("p-divider-vertical", ctx.layout === "vertical")("p-divider-solid", ctx.type === "solid")("p-divider-dashed", ctx.type === "dashed")("p-divider-dotted", ctx.type === "dotted")("p-divider-left", ctx.layout === "horizontal" && (!ctx.align || ctx.align === "left"))("p-divider-center", ctx.layout === "horizontal" && ctx.align === "center" || ctx.layout === "vertical" && (!ctx.align || ctx.align === "center"))("p-divider-right", ctx.layout === "horizontal" && ctx.align === "right")("p-divider-top", ctx.layout === "vertical" && ctx.align === "top")("p-divider-bottom", ctx.layout === "vertical" && ctx.align === "bottom");
        } }, inputs: { style: "style", styleClass: "styleClass", layout: "layout", type: "type", align: "align" }, standalone: true, features: [i0.ɵɵProvidersFeature([DividerStyle]), i0.ɵɵInheritDefinitionFeature, i0.ɵɵStandaloneFeature], ngContentSelectors: _c0, decls: 2, vars: 0, consts: [[1, "p-divider-content"]], template: function Divider_Template(rf, ctx) { if (rf & 1) {
            i0.ɵɵprojectionDef();
            i0.ɵɵelementStart(0, "div", 0);
            i0.ɵɵprojection(1);
            i0.ɵɵelementEnd();
        } }, dependencies: [CommonModule, SharedModule], encapsulation: 2, changeDetection: 0 });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(Divider, [{
        type: Component,
        args: [{
                selector: 'p-divider',
                standalone: true,
                imports: [CommonModule, SharedModule],
                template: `
        <div class="p-divider-content">
            <ng-content></ng-content>
        </div>
    `,
                changeDetection: ChangeDetectionStrategy.OnPush,
                encapsulation: ViewEncapsulation.None,
                host: {
                    '[class.p-divider]': 'true',
                    '[class.p-component]': 'true',
                    '[class.p-divider-horizontal]': 'layout === "horizontal"',
                    '[class.p-divider-vertical]': 'layout === "vertical"',
                    '[class.p-divider-solid]': 'type === "solid"',
                    '[class.p-divider-dashed]': 'type === "dashed"',
                    '[class.p-divider-dotted]': 'type === "dotted"',
                    '[class.p-divider-left]': 'layout === "horizontal" && (!align || align === "left")',
                    '[class.p-divider-center]': '(layout === "horizontal" && align === "center") || (layout === "vertical" && (!align || align === "center"))',
                    '[class.p-divider-right]': 'layout === "horizontal" && align === "right"',
                    '[class.p-divider-top]': 'layout === "vertical" && align === "top"',
                    '[class.p-divider-bottom]': 'layout === "vertical" && align === "bottom"',
                    '[style]': 'inlineStyles',
                    '[attr.aria-orientation]': 'layout',
                    '[attr.data-pc-name]': "'divider'",
                    '[attr.role]': '"separator"',
                    '[style.justifyContent]': 'layout === "horizontal" ? (align === "center" || align === undefined ? "center" : (align === "left" ? "flex-start" : (align === "right" ? "flex-end" : null))) : null',
                    '[style.alignItems]': 'layout === "vertical" ? (align === "center" || align === undefined ? "center" : (align === "top" ? "flex-start" : (align === "bottom" ? "flex-end" : null))) : null'
                },
                providers: [DividerStyle]
            }]
    }], null, { style: [{
            type: Input
        }], styleClass: [{
            type: Input
        }], layout: [{
            type: Input
        }], type: [{
            type: Input
        }], align: [{
            type: Input
        }], hostClass: [{
            type: HostBinding,
            args: ['class']
        }] }); })();
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassDebugInfo(Divider, { className: "Divider" }); })();
export class DividerModule {
    static ɵfac = function DividerModule_Factory(__ngFactoryType__) { return new (__ngFactoryType__ || DividerModule)(); };
    static ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: DividerModule });
    static ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [Divider] });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(DividerModule, [{
        type: NgModule,
        args: [{
                imports: [Divider],
                exports: [Divider]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(DividerModule, { imports: [Divider], exports: [Divider] }); })();
