// primeng/esm2022/inputtext/inputtext.mjs
import { booleanAttribute, Directive, HostListener, inject, Input, NgModule, Optional } from '@angular/core';
import { isEmpty } from '@primeuix/utils';
import { BaseComponent } from 'primeng/basecomponent';
import { InputTextStyle } from './style/inputtextstyle';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
/**
 * InputText directive is an extension to standard input element with theming.
 * @group Components
 */
export class InputText extends BaseComponent {
    ngModel;
    /**
     * Specifies the input variant of the component.
     * @group Props
     */
    variant = 'outlined';
    /**
     * Spans 100% width of the container when enabled.
     * @group Props
     */
    fluid;
    /**
     * Defines the size of the component.
     * @group Props
     */
    pSize;
    filled;
    _componentStyle = inject(InputTextStyle);
    get hasFluid() {
        const nativeElement = this.el.nativeElement;
        const fluidComponent = nativeElement.closest('p-fluid');
        return isEmpty(this.fluid) ? !!fluidComponent : this.fluid;
    }
    constructor(ngModel) {
        super();
        this.ngModel = ngModel;
    }
    ngAfterViewInit() {
        super.ngAfterViewInit();
        this.updateFilledState();
        this.cd.detectChanges();
    }
    ngDoCheck() {
        this.updateFilledState();
    }
    onInput() {
        this.updateFilledState();
    }
    updateFilledState() {
        this.filled = (this.el.nativeElement.value && this.el.nativeElement.value.length) || (this.ngModel && this.ngModel.model);
    }
    static ɵfac = function InputText_Factory(__ngFactoryType__) { return new (__ngFactoryType__ || InputText)(i0.ɵɵdirectiveInject(i1.NgModel, 8)); };
    static ɵdir = /*@__PURE__*/ i0.ɵɵdefineDirective({ type: InputText, selectors: [["", "pInputText", ""]], hostAttrs: [1, "p-inputtext", "p-component"], hostVars: 14, hostBindings: function InputText_HostBindings(rf, ctx) { if (rf & 1) {
            i0.ɵɵlistener("input", function InputText_input_HostBindingHandler($event) { return ctx.onInput($event); });
        } if (rf & 2) {
            i0.ɵɵclassProp("p-filled", ctx.filled)("p-variant-filled", ctx.variant === "filled" || ctx.config.inputStyle() === "filled" || ctx.config.inputVariant() === "filled")("p-inputtext-fluid", ctx.hasFluid)("p-inputtext-sm", ctx.pSize === "small")("p-inputfield-sm", ctx.pSize === "small")("p-inputtext-lg", ctx.pSize === "large")("p-inputfield-lg", ctx.pSize === "large");
        } }, inputs: { variant: "variant", fluid: [2, "fluid", "fluid", booleanAttribute], pSize: "pSize" }, standalone: true, features: [i0.ɵɵProvidersFeature([InputTextStyle]), i0.ɵɵInputTransformsFeature, i0.ɵɵInheritDefinitionFeature] });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(InputText, [{
        type: Directive,
        args: [{
                selector: '[pInputText]',
                standalone: true,
                host: {
                    class: 'p-inputtext p-component',
                    '[class.p-filled]': 'filled',
                    '[class.p-variant-filled]': 'variant === "filled" || config.inputStyle() === "filled" || config.inputVariant() === "filled"',
                    '[class.p-inputtext-fluid]': 'hasFluid',
                    '[class.p-inputtext-sm]': 'pSize === "small"',
                    '[class.p-inputfield-sm]': 'pSize === "small"',
                    '[class.p-inputtext-lg]': 'pSize === "large"',
                    '[class.p-inputfield-lg]': 'pSize === "large"'
                },
                providers: [InputTextStyle]
            }]
    }], () => [{ type: i1.NgModel, decorators: [{
                type: Optional
            }] }], { variant: [{
            type: Input
        }], fluid: [{
            type: Input,
            args: [{ transform: booleanAttribute }]
        }], pSize: [{
            type: Input,
            args: ['pSize']
        }], onInput: [{
            type: HostListener,
            args: ['input', ['$event']]
        }] }); })();
export class InputTextModule {
    static ɵfac = function InputTextModule_Factory(__ngFactoryType__) { return new (__ngFactoryType__ || InputTextModule)(); };
    static ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: InputTextModule });
    static ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({});
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(InputTextModule, [{
        type: NgModule,
        args: [{
                imports: [InputText],
                exports: [InputText]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(InputTextModule, { imports: [InputText], exports: [InputText] }); })();
