// primeng/esm2022/tabs/tabpanels.mjs
import { CommonModule } from '@angular/common';
import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/core';
import { BaseComponent } from 'primeng/basecomponent';
import * as i0 from "@angular/core";
const _c0 = ["*"];
/**
 * TabPanels is a helper component for Tabs component.
 * @group Components
 */
export class TabPanels extends BaseComponent {
    static ɵfac = /*@__PURE__*/ (() => { let ɵTabPanels_BaseFactory; return function TabPanels_Factory(__ngFactoryType__) { return (ɵTabPanels_BaseFactory || (ɵTabPanels_BaseFactory = i0.ɵɵgetInheritedFactory(TabPanels)))(__ngFactoryType__ || TabPanels); }; })();
    static ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: TabPanels, selectors: [["p-tabpanels"]], hostVars: 6, hostBindings: function TabPanels_HostBindings(rf, ctx) { if (rf & 2) {
            i0.ɵɵattribute("data-pc-name", "tabpanels")("role", "presentation");
            i0.ɵɵclassProp("p-tabpanels", true)("p-component", true);
        } }, standalone: true, features: [i0.ɵɵInheritDefinitionFeature, i0.ɵɵStandaloneFeature], ngContentSelectors: _c0, decls: 1, vars: 0, template: function TabPanels_Template(rf, ctx) { if (rf & 1) {
            i0.ɵɵprojectionDef();
            i0.ɵɵprojection(0);
        } }, dependencies: [CommonModule], encapsulation: 2, changeDetection: 0 });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(TabPanels, [{
        type: Component,
        args: [{
                selector: 'p-tabpanels',
                standalone: true,
                imports: [CommonModule],
                template: ` <ng-content></ng-content>`,
                changeDetection: ChangeDetectionStrategy.OnPush,
                encapsulation: ViewEncapsulation.None,
                host: {
                    '[class.p-tabpanels]': 'true',
                    '[class.p-component]': 'true',
                    '[attr.data-pc-name]': '"tabpanels"',
                    '[attr.role]': '"presentation"'
                }
            }]
    }], null, null); })();
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassDebugInfo(TabPanels, { className: "TabPanels" }); })();
