// primeng/esm2022/orderlist/style/orderliststyle.mjs
import { Injectable } from '@angular/core';
import { BaseStyle } from 'primeng/base';
import * as i0 from "@angular/core";
const theme = ({ dt }) => `
.p-orderlist {
    display: flex;
    gap: ${dt('orderlist.gap')};
}

.p-orderlist-controls {
    display: flex;
    flex-direction: column;
    justify-content: center;
    gap: ${dt('orderlist.controls.gap')};
}
`;
const classes = {
    root: 'p-orderlist p-component',
    controls: 'p-orderlist-controls'
};
export class OrderListStyle extends BaseStyle {
    name = 'orderlist';
    theme = theme;
    classes = classes;
    static ɵfac = /*@__PURE__*/ (() => { let ɵOrderListStyle_BaseFactory; return function OrderListStyle_Factory(__ngFactoryType__) { return (ɵOrderListStyle_BaseFactory || (ɵOrderListStyle_BaseFactory = i0.ɵɵgetInheritedFactory(OrderListStyle)))(__ngFactoryType__ || OrderListStyle); }; })();
    static ɵprov = /*@__PURE__*/ i0.ɵɵdefineInjectable({ token: OrderListStyle, factory: OrderListStyle.ɵfac });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(OrderListStyle, [{
        type: Injectable
    }], null, null); })();
/**
 *
 * OrderList is used to maneged the order of a collection.
 *
 * [Live Demo](https://primeng.org/orderlist)
 *
 * @module orderliststyle
 *
 */
export var OrderListClasses;
(function (OrderListClasses) {
    /**
     * Class name of the root element
     */
    OrderListClasses["root"] = "p-orderlist";
    /**
     * Class name of the controls element
     */
    OrderListClasses["controls"] = "p-orderlist-controls";
})(OrderListClasses || (OrderListClasses = {}));
