// primeng/esm2022/inputgroupaddon/inputgroupaddon.mjs
import { CommonModule } from '@angular/common';
import { Component, HostBinding, inject, Input, NgModule } from '@angular/core';
import { SharedModule } from 'primeng/api';
import { BaseComponent } from 'primeng/basecomponent';
import { InputGroupAddonStyle } from './style/inputgroupaddonstyle';
import * as i0 from "@angular/core";
const _c0 = ["*"];
/**
 * InputGroupAddon displays text, icon, buttons and other content can be grouped next to an input.
 * @group Components
 */
export class InputGroupAddon extends BaseComponent {
    /**
     * Inline style of the element.
     * @group Props
     */
    style;
    /**
     * Class of the element.
     * @group Props
     */
    styleClass;
    _componentStyle = inject(InputGroupAddonStyle);
    get hostStyle() {
        return this.style;
    }
    static ɵfac = /*@__PURE__*/ (() => { let ɵInputGroupAddon_BaseFactory; return function InputGroupAddon_Factory(__ngFactoryType__) { return (ɵInputGroupAddon_BaseFactory || (ɵInputGroupAddon_BaseFactory = i0.ɵɵgetInheritedFactory(InputGroupAddon)))(__ngFactoryType__ || InputGroupAddon); }; })();
    static ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: InputGroupAddon, selectors: [["p-inputgroup-addon"], ["p-inputGroupAddon"]], hostVars: 7, hostBindings: function InputGroupAddon_HostBindings(rf, ctx) { if (rf & 2) {
            i0.ɵɵattribute("data-pc-name", "inputgroupaddon");
            i0.ɵɵstyleMap(ctx.hostStyle);
            i0.ɵɵclassMap(ctx.styleClass);
            i0.ɵɵclassProp("p-inputgroupaddon", true);
        } }, inputs: { style: "style", styleClass: "styleClass" }, standalone: true, features: [i0.ɵɵProvidersFeature([InputGroupAddonStyle]), i0.ɵɵInheritDefinitionFeature, i0.ɵɵStandaloneFeature], ngContentSelectors: _c0, decls: 1, vars: 0, template: function InputGroupAddon_Template(rf, ctx) { if (rf & 1) {
            i0.ɵɵprojectionDef();
            i0.ɵɵprojection(0);
        } }, dependencies: [CommonModule], encapsulation: 2 });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(InputGroupAddon, [{
        type: Component,
        args: [{
                selector: 'p-inputgroup-addon, p-inputGroupAddon',
                template: ` <ng-content></ng-content> `,
                standalone: true,
                imports: [CommonModule],
                host: {
                    '[class]': 'styleClass',
                    '[class.p-inputgroupaddon]': 'true',
                    '[attr.data-pc-name]': '"inputgroupaddon"'
                },
                providers: [InputGroupAddonStyle]
            }]
    }], null, { style: [{
            type: Input
        }], styleClass: [{
            type: Input
        }], hostStyle: [{
            type: HostBinding,
            args: ['style']
        }] }); })();
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassDebugInfo(InputGroupAddon, { className: "InputGroupAddon" }); })();
export class InputGroupAddonModule {
    static ɵfac = function InputGroupAddonModule_Factory(__ngFactoryType__) { return new (__ngFactoryType__ || InputGroupAddonModule)(); };
    static ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: InputGroupAddonModule });
    static ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [InputGroupAddon, SharedModule, SharedModule] });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(InputGroupAddonModule, [{
        type: NgModule,
        args: [{
                imports: [InputGroupAddon, SharedModule],
                exports: [InputGroupAddon, SharedModule]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(InputGroupAddonModule, { imports: [InputGroupAddon, SharedModule], exports: [InputGroupAddon, SharedModule] }); })();
