// primeng/esm2022/iftalabel/iftalabel.mjs
import { CommonModule } from '@angular/common';
import { ChangeDetectionStrategy, Component, inject, NgModule, ViewEncapsulation } from '@angular/core';
import { RouterModule } from '@angular/router';
import { SharedModule } from 'primeng/api';
import { BaseComponent } from 'primeng/basecomponent';
import { IftaLabelStyle } from './style/iftalabelstyle';
import * as i0 from "@angular/core";
const _c0 = ["*"];
/**
 * IftaLabel is used to create infield top aligned labels.
 * @group Components
 */
export class IftaLabel extends BaseComponent {
    _componentStyle = inject(IftaLabelStyle);
    static ɵfac = /*@__PURE__*/ (() => { let ɵIftaLabel_BaseFactory; return function IftaLabel_Factory(__ngFactoryType__) { return (ɵIftaLabel_BaseFactory || (ɵIftaLabel_BaseFactory = i0.ɵɵgetInheritedFactory(IftaLabel)))(__ngFactoryType__ || IftaLabel); }; })();
    static ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: IftaLabel, selectors: [["p-iftalabel"], ["p-iftaLabel"], ["p-ifta-label"]], hostAttrs: [1, "p-iftalabel"], standalone: true, features: [i0.ɵɵProvidersFeature([IftaLabelStyle]), i0.ɵɵInheritDefinitionFeature, i0.ɵɵStandaloneFeature], ngContentSelectors: _c0, decls: 1, vars: 0, template: function IftaLabel_Template(rf, ctx) { if (rf & 1) {
            i0.ɵɵprojectionDef();
            i0.ɵɵprojection(0);
        } }, encapsulation: 2, changeDetection: 0 });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(IftaLabel, [{
        type: Component,
        args: [{
                selector: 'p-iftalabel, p-iftaLabel, p-ifta-label',
                standalone: true,
                template: ` <ng-content></ng-content> `,
                changeDetection: ChangeDetectionStrategy.OnPush,
                encapsulation: ViewEncapsulation.None,
                providers: [IftaLabelStyle],
                host: {
                    class: 'p-iftalabel'
                }
            }]
    }], null, null); })();
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassDebugInfo(IftaLabel, { className: "IftaLabel" }); })();
export class IftaLabelModule {
    static ɵfac = function IftaLabelModule_Factory(__ngFactoryType__) { return new (__ngFactoryType__ || IftaLabelModule)(); };
    static ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: IftaLabelModule });
    static ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, SharedModule, RouterModule, SharedModule] });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(IftaLabelModule, [{
        type: NgModule,
        args: [{
                imports: [IftaLabel, CommonModule, SharedModule, RouterModule],
                exports: [IftaLabel, SharedModule]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(IftaLabelModule, { imports: [IftaLabel, CommonModule, SharedModule, RouterModule], exports: [IftaLabel, SharedModule] }); })();
