// primeng/esm2022/picklist/style/pickliststyle.mjs
import { Injectable } from '@angular/core';
import { BaseStyle } from 'primeng/base';
import * as i0 from "@angular/core";
const theme = ({ dt }) => `
.p-picklist {
    display: flex;
    gap: ${dt('picklist.gap')};
}

.p-picklist-controls {
    display: flex;
    flex-direction: column;
    justify-content: center;
    gap: ${dt('picklist.controls.gap')};
}

.p-picklist-list-container {
    flex: 1 1 50%;
}

.p-picklist .p-listbox {
    height: 100%;
}
`;
const classes = {
    root: 'p-picklist p-component',
    sourceControls: 'p-picklist-controls p-picklist-source-controls',
    sourceListContainer: 'p-picklist-list-container p-picklist-source-list-container',
    transferControls: 'p-picklist-controls p-picklist-transfer-controls',
    targetListContainer: 'p-picklist-list-container p-picklist-target-list-container',
    targetControls: 'p-picklist-controls p-picklist-target-controls'
};
export class PickListStyle extends BaseStyle {
    name = 'picklist';
    theme = theme;
    classes = classes;
    static ɵfac = /*@__PURE__*/ (() => { let ɵPickListStyle_BaseFactory; return function PickListStyle_Factory(__ngFactoryType__) { return (ɵPickListStyle_BaseFactory || (ɵPickListStyle_BaseFactory = i0.ɵɵgetInheritedFactory(PickListStyle)))(__ngFactoryType__ || PickListStyle); }; })();
    static ɵprov = /*@__PURE__*/ i0.ɵɵdefineInjectable({ token: PickListStyle, factory: PickListStyle.ɵfac });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PickListStyle, [{
        type: Injectable
    }], null, null); })();
/**
 *
 * PickList is used to reorder items between different lists.
 *
 * [Live Demo](https://www.primeng.org/picklist)
 *
 * @module pickliststyle
 *
 */
export var PickListClasses;
(function (PickListClasses) {
    /**
     * Class name of the root element
     */
    PickListClasses["root"] = "p-picklist";
    /**
     * Class name of the source controls element
     */
    PickListClasses["sourceControls"] = "p-picklist-source-controls";
    /**
     * Class name of the source list container element
     */
    PickListClasses["sourceListContainer"] = "p-picklist-source-list-container";
    /**
     * Class name of the transfer controls element
     */
    PickListClasses["transferControls"] = "p-picklist-transfer-controls";
    /**
     * Class name of the target list container element
     */
    PickListClasses["targetListContainer"] = "p-picklist-target-list-container";
    /**
     * Class name of the target controls element
     */
    PickListClasses["targetControls"] = "p-picklist-target-controls";
})(PickListClasses || (PickListClasses = {}));
