// primeng/esm2022/blockui/style/blockuistyle.mjs
import { Injectable } from '@angular/core';
import { BaseStyle } from 'primeng/base';
import * as i0 from "@angular/core";
const theme = ({ dt }) => `
.p-blockui {
    position: relative;
}

.p-blockui-mask {
    border-radius: ${dt('blockui.border.radius')};
}

.p-blockui-mask.p-overlay-mask {
    position: absolute;
}

.p-blockui-mask-document.p-overlay-mask {
    position: fixed;
}
`;
const classes = {
    root: 'p-blockui'
};
export class BlockUiStyle extends BaseStyle {
    name = 'blockui';
    theme = theme;
    classes = classes;
    static ɵfac = /*@__PURE__*/ (() => { let ɵBlockUiStyle_BaseFactory; return function BlockUiStyle_Factory(__ngFactoryType__) { return (ɵBlockUiStyle_BaseFactory || (ɵBlockUiStyle_BaseFactory = i0.ɵɵgetInheritedFactory(BlockUiStyle)))(__ngFactoryType__ || BlockUiStyle); }; })();
    static ɵprov = /*@__PURE__*/ i0.ɵɵdefineInjectable({ token: BlockUiStyle, factory: BlockUiStyle.ɵfac });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(BlockUiStyle, [{
        type: Injectable
    }], null, null); })();
/**
 *
 * BlockUI represents people using icons, labels and images.
 *
 * [Live Demo](https://www.primeng.org/blockui)
 *
 * @module blockuistyle
 *
 */
export var BlockUIClasses;
(function (BlockUIClasses) {
    /**
     * Class name of the root element
     */
    BlockUIClasses["root"] = "p-blockui";
})(BlockUIClasses || (BlockUIClasses = {}));
