// primeng/esm2022/avatargroup/avatargroup.mjs
import { CommonModule } from '@angular/common';
import { ChangeDetectionStrategy, Component, HostBinding, inject, Input, NgModule, ViewEncapsulation } from '@angular/core';
import { SharedModule } from 'primeng/api';
import { BaseComponent } from 'primeng/basecomponent';
import { AvatarGroupStyle } from './style/avatargroupstyle';
import * as i0 from "@angular/core";
const _c0 = ["*"];
/**
 * AvatarGroup is a helper component for Avatar.
 * @group Components
 */
export class AvatarGroup extends BaseComponent {
    /**
     * Style class of the component
     * @group Props
     */
    styleClass;
    /**
     * Inline style of the component.
     * @group Props
     */
    style;
    get hostClass() {
        return this.styleClass;
    }
    get hostStyle() {
        return this.style;
    }
    _componentStyle = inject(AvatarGroupStyle);
    static ɵfac = /*@__PURE__*/ (() => { let ɵAvatarGroup_BaseFactory; return function AvatarGroup_Factory(__ngFactoryType__) { return (ɵAvatarGroup_BaseFactory || (ɵAvatarGroup_BaseFactory = i0.ɵɵgetInheritedFactory(AvatarGroup)))(__ngFactoryType__ || AvatarGroup); }; })();
    static ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: AvatarGroup, selectors: [["p-avatarGroup"], ["p-avatar-group"], ["p-avatargroup"]], hostVars: 8, hostBindings: function AvatarGroup_HostBindings(rf, ctx) { if (rf & 2) {
            i0.ɵɵstyleMap(ctx.hostStyle);
            i0.ɵɵclassMap(ctx.hostClass);
            i0.ɵɵclassProp("p-avatar-group", true)("p-component", true);
        } }, inputs: { styleClass: "styleClass", style: "style" }, standalone: true, features: [i0.ɵɵProvidersFeature([AvatarGroupStyle]), i0.ɵɵInheritDefinitionFeature, i0.ɵɵStandaloneFeature], ngContentSelectors: _c0, decls: 1, vars: 0, template: function AvatarGroup_Template(rf, ctx) { if (rf & 1) {
            i0.ɵɵprojectionDef();
            i0.ɵɵprojection(0);
        } }, dependencies: [CommonModule, SharedModule], encapsulation: 2, changeDetection: 0 });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(AvatarGroup, [{
        type: Component,
        args: [{
                selector: 'p-avatarGroup, p-avatar-group, p-avatargroup',
                standalone: true,
                imports: [CommonModule, SharedModule],
                template: ` <ng-content></ng-content> `,
                changeDetection: ChangeDetectionStrategy.OnPush,
                encapsulation: ViewEncapsulation.None,
                providers: [AvatarGroupStyle],
                host: {
                    '[class.p-avatar-group]': 'true',
                    '[class.p-component]': 'true'
                }
            }]
    }], null, { styleClass: [{
            type: Input
        }], style: [{
            type: Input
        }], hostClass: [{
            type: HostBinding,
            args: ['class']
        }], hostStyle: [{
            type: HostBinding,
            args: ['style']
        }] }); })();
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassDebugInfo(AvatarGroup, { className: "AvatarGroup" }); })();
export class AvatarGroupModule {
    static ɵfac = function AvatarGroupModule_Factory(__ngFactoryType__) { return new (__ngFactoryType__ || AvatarGroupModule)(); };
    static ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: AvatarGroupModule });
    static ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [AvatarGroup, SharedModule, SharedModule] });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(AvatarGroupModule, [{
        type: NgModule,
        args: [{
                imports: [AvatarGroup, SharedModule],
                exports: [AvatarGroup, SharedModule]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(AvatarGroupModule, { imports: [AvatarGroup, SharedModule], exports: [AvatarGroup, SharedModule] }); })();
