// primeng/esm2022/buttongroup/buttongroup.mjs
import { CommonModule } from '@angular/common';
import { ChangeDetectionStrategy, Component, inject, NgModule, ViewEncapsulation } from '@angular/core';
import { BaseComponent } from 'primeng/basecomponent';
import { ButtonGroupStyle } from './style/buttongroupstyle';
import * as i0 from "@angular/core";
const _c0 = ["*"];
export class ButtonGroup extends BaseComponent {
    _componentStyle = inject(ButtonGroupStyle);
    static ɵfac = /*@__PURE__*/ (() => { let ɵButtonGroup_BaseFactory; return function ButtonGroup_Factory(__ngFactoryType__) { return (ɵButtonGroup_BaseFactory || (ɵButtonGroup_BaseFactory = i0.ɵɵgetInheritedFactory(ButtonGroup)))(__ngFactoryType__ || ButtonGroup); }; })();
    static ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: ButtonGroup, selectors: [["p-buttonGroup"], ["p-buttongroup"], ["p-button-group"]], standalone: true, features: [i0.ɵɵProvidersFeature([ButtonGroupStyle]), i0.ɵɵInheritDefinitionFeature, i0.ɵɵStandaloneFeature], ngContentSelectors: _c0, decls: 2, vars: 0, consts: [["role", "group", 1, "p-buttongroup", "p-component"]], template: function ButtonGroup_Template(rf, ctx) { if (rf & 1) {
            i0.ɵɵprojectionDef();
            i0.ɵɵelementStart(0, "span", 0);
            i0.ɵɵprojection(1);
            i0.ɵɵelementEnd();
        } }, dependencies: [CommonModule], encapsulation: 2, changeDetection: 0 });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(ButtonGroup, [{
        type: Component,
        args: [{
                selector: 'p-buttonGroup, p-buttongroup, p-button-group',
                standalone: true,
                imports: [CommonModule],
                template: `
        <span class="p-buttongroup p-component" role="group">
            <ng-content></ng-content>
        </span>
    `,
                changeDetection: ChangeDetectionStrategy.OnPush,
                encapsulation: ViewEncapsulation.None,
                providers: [ButtonGroupStyle]
            }]
    }], null, null); })();
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassDebugInfo(ButtonGroup, { className: "ButtonGroup" }); })();
export class ButtonGroupModule {
    static ɵfac = function ButtonGroupModule_Factory(__ngFactoryType__) { return new (__ngFactoryType__ || ButtonGroupModule)(); };
    static ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: ButtonGroupModule });
    static ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [ButtonGroup] });
}
(() => { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(ButtonGroupModule, [{
        type: NgModule,
        args: [{
                imports: [ButtonGroup],
                exports: [ButtonGroup]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(ButtonGroupModule, { imports: [ButtonGroup], exports: [ButtonGroup] }); })();
